src/engine/execute.js

```diff
@@ -21,9 +30,28 @@ var execute = function (sequencer, thread) {
     }
 
     var blockFunction = runtime.getOpcodeFunction(opcode);
-    if (!blockFunction) {
-        console.warn('Could not get implementation for opcode: ' + opcode);
-        return;
+    var isHat = runtime.getIsHat(opcode);
+
+    // Hats are implemented slightly differently from regular blocks.
+    // If they have an associated block function, it's treated as a predicate;
+    // if not, execution will proceed right through it (as a no-op).
+    if (isHat) {
+        var nextBlock = target.blocks.getNextBlock(currentBlockId);
+        if (!nextBlock) {
+            // Hat with no next block - don't try to evaluate it.
+            sequencer.retireThread(thread);
+            return;
```

Comment:
If custom reporters were to become a thing, there would most likely be side-effecting custom reporters, e.g. an array pop implementation. That means that something such as this:

![scratchblocks](https://cloud.githubusercontent.com/assets/9948030/17930895/e9e184de-69de-11e6-8751-5e00c863c488.png)

..would never trigger the "pop" block.

Not really a huge issue but probably worth noting!